babeld/message.c

```diff
@@ -288,13 +288,18 @@ channels_len(unsigned char *channels)
 static int
 babel_packet_examin(const unsigned char *packet, int packetlen)
 {
-    unsigned i = 0, bodylen;
+    int i = 0, bodylen;
     const unsigned char *message;
     unsigned char type, len;
 
     if(packetlen < 4 || packet[0] != 42 || packet[1] != 2)
         return 1;
     DO_NTOHS(bodylen, packet + 2);
+    if(bodylen + 4 > packetlen) {
+        flog_err(EC_BABEL_PACKET, "Received truncated packet (%d + 4 > %d).",
```

Comment:
why log an error here, when the other truncated paths are just debugs?

Reply:
This follows the existing code as below:

https://github.com/FRRouting/frr/blob/3d1ff4bfdb159583a62e72ccb359a8508c4cbb2e/babeld/message.c#L368-L374

Reply:
I'm just pointing out that the other three cases in this same function only debug, and don't log?

Reply:
> I'm just pointing out that the other three cases in this same function only debug, and don't log?

If you like to use debug, I can just change it. Do you think it is necessary to change it to debug?

Reply:
Yes, please, change it to debug.
And why do you do `bodylen = packetlen - 4;` and continue processing? We received invalid packet, so we must stop processing and drop it.

Reply:
> Yes, please, change it to debug. And why do you do `bodylen = packetlen - 4;` and continue processing? We received invalid packet, so we must stop processing and drop it.

Thanks for your reply! Sure, I will change it to debug.

I do `bodylen = packetlen - 4` because I follow the existing code at Line 370 of `message.c` in the master branch, which also does this operation without dropping the invalid packet. Please check the code at Line 370 and confirm with me we need to drop it.

Reply:
Yes, please, change the code to drop the packet. How can we continue processing it, if the header says it needs a specific amount of bytes and we don't have them?

Reply:
> Yes, please, change the code to drop the packet. How can we continue processing it, if the header says it needs a specific amount of bytes and we don't have them?

Done. I have changed the code to drop the invalid packet.